common/lanelet2_extension/test/src/test_utilities.cpp

```diff
@@ -137,6 +239,93 @@ TEST_F(TestSuite1, OverwriteLaneletsCenterline)
+  ASSERT_EQ(sample_map_ptr->laneletLayer.find(road_lanelet.id())->regulatoryElements().size(),1);
+  ASSERT_EQ(sample_map_ptr->laneletLayer.find(road_lanelet.id())->regulatoryElements()[0]->id(),pcl->id());
+  rf = lanelet::utils::query::findReferences(tl, sample_map_ptr);
+  ASSERT_EQ(rf.regems.size(), 0); 
+  ASSERT_EQ(rf.lss.size(), 3); //tl local copy still references these 3 parameters, 
+                              // but those parameters don't reference it back as it should 
+  ASSERT_EQ(rf.llts.size(), 0); // connection with parent llt is severed
+
+  // Test if map is valid by writing it to a file and loading it
+  // Build new map from modified data
+  std::vector<lanelet::Area> new_areas;
+  std::vector<lanelet::Lanelet> new_lanelets;
+  for (auto lanelet : sample_map_ptr->laneletLayer)
+  {
+    std::cerr << "Adding lanelet: " << lanelet.id() << std::endl;
+    lanelet::Lanelet mutable_ll = sample_map_ptr->laneletLayer.get(lanelet.id());
+    new_lanelets.emplace_back(mutable_ll);
+  }
+  for (auto area : sample_map_ptr->areaLayer)
+  {
+    std::cerr << "Adding area: " << area.id() << std::endl;
+    lanelet::Area mutable_area = sample_map_ptr->areaLayer.get(area.id());
+    new_areas.emplace_back(mutable_area);
+  }
+  auto new_map = lanelet::utils::createMap(new_lanelets, new_areas);
+  // Write new map to file
+  std::string new_file = "modified_map.osm";
+  lanelet::projection::LocalFrameProjector local_projector("+proj=tmerc +lat_0=39.46636844371259 +lon_0=-76.16919523566943 +k=1 +x_0=0 +y_0=0 +datum=WGS84 +units=m +vunits=m +no_defs");
+  lanelet::ErrorMessages write_errors;
+
+  lanelet::write(new_file, *new_map, local_projector, &write_errors);
+
+  if (write_errors.size() > 0)
+  {
+    std::cerr << "Errors occurred while writing the map! Output file located at " << new_file << std::endl;
+  }
+  else
+  {
+    std::cerr << "Map written without errors to: " << new_file << std::endl;
+  }
+  for (auto msg : write_errors)
+  {
+    std::cerr << "Write Error: " << msg << std::endl;
+  }
+
+  // Now try to load it back
+  using namespace lanelet;
+  LaneletMapPtr map = lanelet::load(new_file, local_projector);
+  lanelet::validation::ValidationConfig config;
+  auto issues = lanelet::validation::validateMap(*map, config);
+  auto report = lanelet::validation::buildReport(issues);
+  EXPECT_EQ(13, report.warnings.size()); // these 13 warnings have been verified  
+                                         // that they are not related to our issue of focus
+  EXPECT_EQ(0ul, report.errors.size());
+  // In fact, it should be routable
+  traffic_rules::TrafficRulesPtr trafficRules =
+    traffic_rules::TrafficRulesFactory::create(Locations::Germany, Participants::Vehicle);
+  auto new_routing_graph = lanelet::routing::RoutingGraph::build(*map, *trafficRules);
+  new_routing_graph->exportGraphViz("modified_map_graph.viz"); //TODO: not sure where it is writing to...
```

Comment:
If you add a 
WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/test
To and test field in CMakeList.txt is will set the working directory to that folder. Then you can use a relative path to specify the file location. 

But since this line is not actually used in the unit test you could just comment it out or delete it.